requests_oauthlib/oauth2_session.py

```diff
@@ -192,23 +209,66 @@ def fetch_token(self, token_url, code=None, authorization_response=None,
                 raise ValueError('Please supply either code or '
                                  'authorization_response parameters.')
 
+        # Earlier versions of this library build an HTTPBasicAuth header out of
+        # `username` and `password`. The RFC states, however these attributes
+        # must be in the request body and not the header.
+        # If an upstream server is not spec compliant and requires them to
+        # appear as an Authorization header, supply an explicit `auth` header
+        # to this function.
+        # This check will allow for empty strings, but not `None`.
+        #
+        # Refernences
+        # 4.3.2 - Resource Owner Password Credentials Grant
+        #         https://tools.ietf.org/html/rfc6749#section-4.3.2
+
+        if isinstance(self._client, LegacyApplicationClient):
+            if username is None:
+                raise ValueError('`LegacyApplicationClient` requires both the '
+                                 '`username` and `password` parameters.')
+            if password is None:
+                raise ValueError('The required paramter `username` was supplied, '
+                                 'but `password` was not.')
+
+        # TODO:
```

Comment:
As discussed in https://github.com/oauthlib/oauthlib/issues/585#issuecomment-421169638, `username` and `password` are used only for `LegacyApplicationClient`. However, we should not *restrict* or try to *remove* them from the params, if they are present. But we should not do any processing either, since they are not described in the RFC (for those other grants!).

Reply:
Great. so just remove the `TODO` line that I put in for code review - correct?

Reply:
Yes IMHO

Reply:
done.

the way i was notified of this by code-highlighting, I thought there may have been another issue or mis-interpretation. I just wanted to ensure we are both talking about the TODO line.